### PR TITLE
[bitnami/airflow] Add VIB tests

### PR DESCRIPTION
**Pull request by mdhont**
### Description of the change

The main objective of this PR is to publish our Bitnami airflow container using VMware Image Builder. In order to do that, several changes are included:

- Increasing the existing test coverage of the asset by adding Goss tests.
- Update verify and publish VIB pipeline's definitions.

### Benefits

- Ensuring higher quality of the container catalog.
- Increased pool of assets completely handled by VMware Image Builder.

### Possible drawbacks

- Automated tests could introduce additional flakiness to the CI/CD.

### Applicable issues

NA

Last successful test run:

https://github.com/bitnami/containers/actions/runs/4915826874/jobs/8778800701

### CODE REVIEW

File: .vib/airflow/goss/airflow.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    exit-status: 0
+    stdout:
+      - 0
+file:
+  /opt/bitnami/airflow/venv/lib/python3.9/site-packages/airflow/www/node_modules:
+    exists: false
+  /opt/bitnami/airflow/venv/lib/python3.9/site-packages/airflow/config_templates/default_airflow.cfg:
+    exists: true
+    contains:
+      - Typically, this is a simple statement like "SELECT 1"
```
Comment (joancafom) [inline .vib/airflow/goss/airflow.yaml:20]: This is no longer relevant, the issue this refers to was solved in upstream:

https://github.com/apache/airflow/blob/main/airflow/config_templates/default_airflow.cfg#L307

In fact, you should create an internal task to solve it

File: .vib/airflow/goss/airflow.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    exec: airflow users list
+    exit-status: 0
+    stdout:
+      - No data found
+  check-so-files:
+    # Check that all .so files in site-packages are executable
+    exec: find /opt/bitnami/airflow/venv/lib/python3.9/site-packages/ -type f -name '*.so.*' ! -executable | wc -l
+    exit-status: 0
+    stdout:
+      - 0
```
Comment (joancafom) [inline .vib/airflow/goss/airflow.yaml:13]: I don't think this adds much value. Those files are third party packages...

**Review by joancafom — COMMENTED** [review]
Thanks for your PR! I left you some comments as a review

File: .vib/airflow/goss/airflow.yaml (hunk 2 above)
Comment (joancafom) [reply .vib/airflow/goss/airflow.yaml:13]: Instead, you could list the `venv` for submodules that should've been installed, such as `celery`, `docker`, `cncf.kubernetes` or `hashicorp`

File: .vib/airflow/goss/airflow.yaml (hunk 1 above)
Comment (mdhont) [reply .vib/airflow/goss/airflow.yaml]: I will create a task

File: .vib/airflow/goss/airflow.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      - /microsoft.azure/
+      - mysql
+      - postgres
+      - redis
+      - sendgrid
+      - sftp
+      - slack
+      - ssh
+      - statsd
+      - virtualenv
```
Comment (joancafom) [inline .vib/airflow/goss/airflow.yaml:36]: I would not check this whole list of modules, just a subset of the more relevant ones:

```suggestion
      - async
      - amazon
      - celery
      - /cncf.kubernetes/
      - docker
      - hashicorp
      - ldap
      - google
      - /microsoft.azure/
      - mysql
      - postgres
      - redis
```

Comment (mdhont) [reply .vib/airflow/goss/airflow.yaml]: I've also moved the list to the vars.yaml, based on earlier comments from other diffs.

File: .vib/airflow/vib-publish.json [hunk 4]
```diff
@@ -18,7 +19,8 @@
               }
             },
             "architectures": [
-              "linux/amd64"
+              "linux/amd64",
+              "linux/arm64"
```
[the remark this answers is not in the document]
Comment (mdhont) [reply .vib/airflow/vib-publish.json]: I usually copy from another file, this one slipped through.

**Review by agomezmoron — APPROVED** [review]

**Review by joancafom — APPROVED** [review]

**Pull request merged**